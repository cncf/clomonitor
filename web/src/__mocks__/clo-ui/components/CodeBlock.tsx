import { FunctionComponent, ReactNode } from 'react';

type CodeBlockProps = {
  content: ReactNode;
  label?: string;
  withCopyBtn?: boolean;
};

const CodeBlock: FunctionComponent<CodeBlockProps> = ({ content, label, withCopyBtn }) => {
  return (
    <div data-testid="code" data-label={label}>
      {content}
      {withCopyBtn && label ? (
        <button aria-label={label} type="button">
          {label}
        </button>
      ) : null}
    </div>
  );
};

export { CodeBlock };
