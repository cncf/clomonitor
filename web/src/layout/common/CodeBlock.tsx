import { useContext } from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { docco } from 'react-syntax-highlighter/dist/cjs/styles/hljs';
import { tomorrowNight } from 'react-syntax-highlighter/dist/cjs/styles/hljs';

import { AppContext } from '../../context/AppContextProvider';
import ButtonCopyToClipboard from './ButtonCopyToClipboard';
import styles from './CodeBlock.module.css';

interface Props {
  language: string;
  content: string;
  label: string;
  styles?: { [key: string]: string };
}

const CodeBlock = (props: Props) => {
  const { ctx } = useContext(AppContext);
  const { effective } = ctx.prefs.theme;
  const isDarkActive = effective === 'dark';

  return (
    <div className={`d-flex flex-row align-items-center pb-2 ${styles.codeBlock}`}>
      <SyntaxHighlighter
        language={props.language}
        style={isDarkActive ? tomorrowNight : docco}
        customStyle={{
          backgroundColor: 'var(--bg-code)',
          color: 'var(--color-font)',
          padding: '1rem 0.5rem',
          marginBottom: 0,
          ...props.styles,
        }}
      >
        {props.content}
      </SyntaxHighlighter>

      <ButtonCopyToClipboard text={props.content} label={props.label} wrapperClassName="ms-3" />
    </div>
  );
};

export default CodeBlock;
