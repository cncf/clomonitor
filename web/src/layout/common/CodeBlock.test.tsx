import { render, screen } from '@testing-library/react';

import CodeBlock from './CodeBlock';

const defaultProps = {
  language: 'markdown',
  content: '##Sample',
  label: 'Copy btn',
};

describe('CodeBlock', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<CodeBlock {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<CodeBlock {...defaultProps} />);

    const code = screen.getByTestId('code');
    expect(code).toBeInTheDocument();
    expect(code).toHaveTextContent('##Sample');

    expect(screen.getByRole('button', { name: 'Copy btn' })).toBeInTheDocument();
  });
});
