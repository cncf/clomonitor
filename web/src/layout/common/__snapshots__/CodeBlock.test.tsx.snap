// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-row align-items-center pb-2 codeBlock"
    data-testid="code"
  >
    <pre
      style="display: block; overflow-x: auto; padding: 1rem; background: rgb(248, 248, 255); margin-bottom: 0px; width: calc(100% - 1rem - 32px);"
    >
      <code
        class="language-markdown"
        style="white-space: pre;"
      >
        <span
          style="color: rgb(25, 70, 157);"
        >
          ##Sample
        </span>
      </code>
    </pre>
    <div
      class="position-relative ms-3"
    >
      <button
        aria-label="Copy btn"
        class="btn btn-sm btn-primary rounded-circle btn"
        type="button"
      >
        <div
          aria-hidden="true"
          class="d-flex flex-row align-items-center"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="13"
              rx="2"
              ry="2"
              width="13"
              x="9"
              y="9"
            />
            <path
              d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
